Scripts/datatypes/purpose.py

```diff
@@ -583,3 +583,26 @@ def calc_traffic(self, impedance: dict, purpose_key: str):
         generation = numpy.tile(self.zone_data[f"gen_{purpose_key}"], (nr_zones, 1))
         demand = {mode: (probs.pop(mode) * generation).T for mode in self.modes}
         return demand
+
+    def calc_vehicles(self, matrix: numpy.ndarray, ass_class: str, args: dict):
+        """Calculate vehicle matrix from ton matrix using ton-to-vehicles 
+        conversion values.
+
+        Parameters
+        ----------
+        matrix : numpy.ndarray
+            ton matrix
+        ass_class : str
+            truck assignment class
+        args : dict
+            conversion variable : float
```

Comment:
This seems to be purpose-specific, so I do not see the point in feeding it from the outside here.